src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNet.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Hosting.Internal
+{
+    internal static class HostingLoggerExtensions
+    {
+        public static IDisposable RequestScope(this ILogger logger, HttpContext httpContext)
+        {
+            return logger.BeginScopeImpl(new HostingLogScope(httpContext));
+        }
+
+        public static void RequestStarting(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(
+                    logLevel: LogLevel.Information,
+                    eventId: 1,
+                    state: new HostingRequestStarting(httpContext),
+                    exception: null,
+                    formatter: HostingRequestStarting.Callback);
+            }
+        }
+
+        public static void RequestFinished(this ILogger logger, HttpContext httpContext)
+        {
+            if (logger.IsEnabled(LogLevel.Information))
+            {
+                logger.Log(
+                    logLevel: LogLevel.Information,
+                    eventId: 2,
+                    state: new HostingRequestFinished(httpContext),
+                    exception: null,
+                    formatter: HostingRequestFinished.Callback);
+            }
+        }
+
+        private class HostingLogScope : ILogValues
+        {
+            private readonly HttpContext _httpContext;
```

Comment:
Doesn't it go via `Microsoft.Extensions.Logging.Logger` which enumerates all loggers in the `IsEnabled` method bailing on first true?

Also have fast path PR for no loggers for the check: https://github.com/aspnet/Logging/pull/254/files

An `IsDisabled` with no params would be prettier, but would change the interface.